### ISSUE TITLE
[BUG] Infinite rerender loop when tooltip is open for an anchor that is inside a scrolled overflow

### ISSUE THREAD

**Issue by johannkor** [ReactTooltip/react-tooltip#1146]
**Bug description**
When you open a tooltip whose anchor is inside a scrollable container (with `overflow:scroll`), if you scroll the container so that the anchor is hidden inside the overflow, `Tooltip` goes into a computePosition rerender loop.

**Version of Package**
v5.25.1

- `@floating-ui/dom`: `1.5.1`

**To Reproduce**
1. Clone this repo
2. Replace `src/index-dev.ts` with this:
```tsx
import React, { useRef } from 'react'
import ReactDOM from 'react-dom'
import { TooltipController as Tooltip } from 'components/TooltipController'
import { TooltipRefProps } from 'components/Tooltip/TooltipTypes'
import './tokens.css'

const ANCHOR_ID = 'anchor'

function App() {
  const tooltipRef1 = useRef<TooltipRefProps>(null)

  return (
    <>
      <div
        style={{
          display: 'flex',
          alignItems: 'flex-start',
        }}
      >
        <main
          style={{
            maxHeight: '50vh',
            overflowY: 'scroll',
          }}
        >
          <div style={{ width: '300px', border: '1px solid gray' }}>
            <span id={ANCHOR_ID} style={{ border: '1px solid red', fontSize: '30px' }}>
              anchor
            </span>
            {/* force overflow so we have something to scroll */}
            <div style={{ height: '100vh' }}></div>
          </div>
        </main>

        <aside>
          <button
            onClick={(e) => {
              e.preventDefault()
              e.stopPropagation()

              tooltipRef1.current?.open({
                anchorSelect: `#${ANCHOR_ID}`,
                content: 'Hello world!',
              })
            }}
          >
            open tooltip
          </button>
        </aside>
      </div>

      <Tooltip ref={tooltipRef1} />
    </>
  )
}

ReactDOM.render(<App />, document.getElementById('app'))
```
3. Run `yarn` and `yarn dev`
4. Open the dev server in a browser
5. Click on the button to open the tooltip
6. Scroll the left container down so that the anchor goes up and gets hidden
7. Open console, see that Tooltip is now rerendering constantly (place a logpoint or console.log() inside `Tooltip.tsx`'s render)
8. Scroll down so that the anchor is visible
9. Notice that it now only rerenders when it needs to (e.g. scroll changes)

**Expected behavior**
I would expect it to only update the position when necessary

**Screenshots**

**Desktop (please complete the following information if possible or delete this section):**
 - OS: macOS Sonoma
 - Browser: Chrome 120
 - Frameworks: React 16, React 18 (non-concurrent mode)

**Additional context**
- Split from #1144
- It appears that `positionStrategy="fixed"` does not resolve this issue

**Comment by gabrieljablonski** [comment]
Beta version `react-tooltip@5.26.1-beta.1169.0` (from #1169) improves how the tooltip handles some internal states, which seems to fix this.

Feel free to test test it out until we release an official version.

**Comment by gabrieljablonski** [comment]
Fixed on official release `react-tooltip@5.26.2`
